src/Symfony/Component/Notifier/Bridge/OneSignal/OneSignalTransport.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\OneSignal;
+
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\PushMessage;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Tomas Norkūnas <norkunas.tom@gmail.com>
+ */
+final class OneSignalTransport extends AbstractTransport
+{
+    protected const HOST = 'onesignal.com';
+
+    private $appId;
+    private $apiKey;
+    private $recipientId;
+
+    public function __construct(string $appId, string $apiKey, ?string $recipientId = null, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
```

Comment:
I don't think that `$recipientId` should be provided here, because this transport may be used to send multiple different messages during its lifetime.

Unless you want this to be a default recipient for cases when none are provided - in that case, I think this should be renamed to `$defaultRecipientId` to clarify its intent.